include/vcpkg/paragraphs.h

```diff
@@ -27,22 +27,26 @@ namespace vcpkg::Paragraphs
     bool is_port_directory(const ReadOnlyFilesystem& fs, const Path& maybe_directory);
 
     // If an error occurs, the Expected will be in the error state.
-    // Otherwise, if the port is known, the unique_ptr contains the loaded port information.
-    // Otherwise, the unique_ptr is nullptr.
-    ExpectedL<std::unique_ptr<SourceControlFile>> try_load_port(const ReadOnlyFilesystem& fs,
-                                                                StringView port_name,
-                                                                const Path& port_directory);
+    // Otherwise, if the port is known, result->source_control_file contains the loaded port information.
+    // Otherwise, result->source_control_file is nullptr.
+    ExpectedL<SourceControlFileAndLocation> try_load_port(const ReadOnlyFilesystem& fs,
+                                                          StringView port_name,
+                                                          const PortLocation& port_location);
     // Identical to try_load_port, but the port unknown condition is mapped to an error.
-    ExpectedL<std::unique_ptr<SourceControlFile>> try_load_port_required(const ReadOnlyFilesystem& fs,
-                                                                         StringView port_name,
-                                                                         const Path& port_directory);
-    ExpectedL<std::unique_ptr<SourceControlFile>> try_load_project_manifest_text(StringView text,
-                                                                                 StringView origin,
-                                                                                 MessageSink& warning_sink);
-    ExpectedL<std::unique_ptr<SourceControlFile>> try_load_port_manifest_text(StringView text,
-                                                                              StringView origin,
-                                                                              MessageSink& warning_sink);
-    ExpectedL<std::unique_ptr<SourceControlFile>> try_load_control_file_text(StringView text, StringView origin);
+    ExpectedL<SourceControlFileAndLocation> try_load_port_required(const ReadOnlyFilesystem& fs,
+                                                                   StringView port_name,
+                                                                   const PortLocation& port_location);
+    ExpectedL<SourceControlFileAndLocation> try_load_project_manifest_text(StringView text,
+                                                                           StringView origin,
```

Comment:
This should be `control_location`, or `SCFL`'s member should be called `origin`.

Reply:
Also, consider introducing:

```c++
struct ControlFileLocation {
Path control_location;
std:string spdx_location;
};
```